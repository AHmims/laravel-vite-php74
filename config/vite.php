<?php

return [
    /*
    |--------------------------------------------------------------------------
    | Configurations
    |--------------------------------------------------------------------------
    | The following describes a set of configurations that can be used
    | independently. Because Vite does not support generating multiple
    | bundles, using separate configuration files is necessary.
    */
    'configs' => [
        'default' => [
            'entrypoints' => [
                'ssr' => 'resources/scripts/ssr.ts',
                'paths' => [
                    'resources/scripts/main.ts',
                    'resources/js/app.js',
                ],
                'ignore' => '/\\.(d\\.ts|json)$/',
            ],
            'dev_server' => [
                'enabled' => true,
                'url' => env('DEV_SERVER_URL', 'http://localhost:3000'),
                'ping_before_using_manifest' => true,
                'ping_url' => null,
                'ping_timeout' => 1,
                'key' => env('DEV_SERVER_KEY'),
                'cert' => env('DEV_SERVER_CERT'),
            ],
            'build_path' => 'build',
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Aliases
    |--------------------------------------------------------------------------
    | You can define aliases to avoid having to make relative imports.
    | Aliases will be written to tsconfig.json automatically so your IDE
    | can know how to resolve them.
    */
    'aliases' => [
        '@' => 'resources',
    ],

    /*
    |--------------------------------------------------------------------------
    | Commands
    |--------------------------------------------------------------------------
    | Before starting the development server or building the assets, you
    | may need to run specific commands. With these options, you can
    | define what to run, automatically.
    */
    'commands' => [
        'artisan' => [
            // 'typescript:generate'
        ],
        'shell' => [
            //
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Testing
    |--------------------------------------------------------------------------
    | While testing, you may need to use the manifest, or not, depending
    | what and how you test. You can tune these options according to
    | your needs.
    */
    'testing' => [
        'use_manifest' => false,
    ],

    /*
    |--------------------------------------------------------------------------
    | Default interfaces
    |--------------------------------------------------------------------------
    | Here you may change how some parts of the package work by replacing
    | their associated logic.
    */
    'server_checker' => Innocenzi\Vite\ServerCheckers\HttpServerChecker::class,
    'tag_generator' => Innocenzi\Vite\TagGenerators\DefaultTagGenerator::class,
    'entrypoints_finder' => Innocenzi\Vite\EntrypointsFinder\DefaultEntrypointsFinder::class,

    /*
    |--------------------------------------------------------------------------
    | Default configuration
    |--------------------------------------------------------------------------
    | Here you may specify which of the configurations above you wish
    | to use as your default one.
    */
    'default' => env('VITE_DEFAULT_CONFIG', 'default'),
];
