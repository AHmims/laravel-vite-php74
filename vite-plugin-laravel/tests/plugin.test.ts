import path from 'node:path'
import { expect, it } from 'vitest'
import { build } from 'vite'
import laravel from '../src'

it.only('runs the plugin as expected', async() => {
	const result = await build({
		root: path.resolve(__dirname, '__fixtures__'),
		logLevel: 'silent',
		plugins: [
			laravel({
				config: {
					aliases: { '@': 'vite-plugin-laravel/tests/__fixtures__/resources' },
					build_path: 'build',
					dev_server: { url: 'http://localhost:3000' },
					entrypoints: { paths: 'resources/scripts/app.ts' },
				},
			}),
		],
	})

	expect(result).toMatchSnapshot()
})
