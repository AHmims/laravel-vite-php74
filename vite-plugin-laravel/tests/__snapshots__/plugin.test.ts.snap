// Vitest Snapshot v1

exports[`runs the plugin as expected 1`] = `
{
  "output": [
    {
      "code": "console.log(\\"app content\\");
",
      "dynamicImports": [],
      "exports": [],
      "facadeModuleId": "D:/Code/packages/laravel-vite/vite-plugin-laravel/tests/__fixtures__/resources/scripts/app.ts",
      "fileName": "assets/app.d997abbd.js",
      "implicitlyLoadedBefore": [],
      "importedBindings": {},
      "imports": [],
      "isDynamicEntry": false,
      "isEntry": true,
      "isImplicitEntry": false,
      "map": null,
      "modules": {
        "D:/Code/packages/laravel-vite/vite-plugin-laravel/tests/__fixtures__/resources/css/styles.css": {
          "code": "var styles = '';",
          "originalLength": 22,
          "removedExports": [],
          "renderedExports": [
            "default",
          ],
          "renderedLength": 16,
        },
        "D:/Code/packages/laravel-vite/vite-plugin-laravel/tests/__fixtures__/resources/scripts/app.ts": {
          "code": "console.log(\\"app content\\");",
          "originalLength": 53,
          "removedExports": [],
          "renderedExports": [],
          "renderedLength": 27,
        },
      },
      "name": "app",
      "referencedFiles": [],
      "type": "chunk",
    },
    {
      "fileName": "assets/app.59db6eb8.css",
      "isAsset": true,
      "name": "app.css",
      "source": "body{color:red}
",
      "type": "asset",
    },
    {
      "fileName": "manifest.json",
      "isAsset": true,
      "name": undefined,
      "source": "{
  \\"resources/scripts/app.ts\\": {
    \\"file\\": \\"assets/app.d997abbd.js\\",
    \\"src\\": \\"resources/scripts/app.ts\\",
    \\"isEntry\\": true,
    \\"css\\": [
      \\"assets/app.59db6eb8.css\\"
    ]
  }
}",
      "type": "asset",
    },
  ],
}
`;
